examples/standalone/02_hello_window/src/main.rs

```diff
@@ -0,0 +1,169 @@
+                },
+                None, // Trace path
+            )
+            .await
+            .unwrap();
+
+        State {
+            surface,
+            device,
+            adapter,
+            queue,
+            size,
+        }
+    }
+
+    fn resize(&mut self, new_size: winit::dpi::PhysicalSize<u32>) {
+        self.size = new_size;
+        self.render();
+    }
+
+    fn render(&mut self) {
+        // Configure the surface
+        let cap = self.surface.get_capabilities(&self.adapter);
+        let surface_config = wgpu::SurfaceConfiguration {
+            usage: wgpu::TextureUsages::RENDER_ATTACHMENT,
+            format: cap.formats[0],
+            view_formats: vec![cap.formats[0]],
+            alpha_mode: wgpu::CompositeAlphaMode::Auto,
+            width: self.size.width,
+            height: self.size.height,
+            desired_maximum_frame_latency: 2,
+            present_mode: wgpu::PresentMode::Mailbox,
+        };
+        self.surface.configure(&self.device, &surface_config);
+
+        // Create texture view
+        let surface_texture = self
+            .surface
+            .get_current_texture()
+            .expect("failed to acquire next swapchain texture");
+        let texture_view = surface_texture
+            .texture
+            .create_view(&wgpu::TextureViewDescriptor::default());
+
+        // Renders a GREEN screen
+        let mut encoder = self.device.create_command_encoder(&Default::default());
+        {
+            let _renderpass = encoder.begin_render_pass(&wgpu::RenderPassDescriptor {
+                label: None,
+                color_attachments: &[Some(wgpu::RenderPassColorAttachment {
+                    view: &texture_view,
+                    resolve_target: None,
+                    ops: wgpu::Operations {
+                        load: wgpu::LoadOp::Clear(wgpu::Color::GREEN),
+                        store: wgpu::StoreOp::Store,
+                    },
+                })],
+                depth_stencil_attachment: None,
+                timestamp_writes: None,
+                occlusion_query_set: None,
+            });
+        }
+
+        // Submit the command in the queue to execute
+        self.queue.submit(Some(encoder.finish()));
+        surface_texture.present();
+    }
+}
+
+#[derive(Default)]
+struct App {
+    window: Option<Arc<Window>>,
+    state: Option<State>,
+}
+
+impl ApplicationHandler for App {
+    fn resumed(&mut self, event_loop: &ActiveEventLoop) {
+        // Create window object
+        let window = Arc::new(
+            event_loop
+                .create_window(Window::default_attributes())
+                .unwrap(),
+        );
+
+        let mut state = pollster::block_on(async { State::new(window.clone()).await });
+        state.render();
+
+        self.window = Some(window);
+        self.state = Some(state);
+    }
+
+    fn window_event(&mut self, event_loop: &ActiveEventLoop, _id: WindowId, event: WindowEvent) {
```

Comment:
I think we should be constantly rendering in these examples and mention how to do on-demand rendering in a comment.